jdaviz/configs/specviz2d/plugins/spectral_extraction/tests/test_spectral_extraction.py

```diff
@@ -72,3 +72,14 @@ def test_plugin(specviz2d_helper):
     # create subtracted spectrum
     sp_ext = pext.create_extract()
     assert isinstance(sp_ext, Spectrum1D)
+
+
+@pytest.mark.remote_data
+def test_spectrum_on_top(specviz2d_helper):
+    fn = download_file('https://mast.stsci.edu/api/v0.1/Download/file/?uri=mast:jwst/product/jw01529-c1002_t002_miri_p750l_s2d.fits', cache=True)  # noqa
+
+    specviz2d_helper.load_data(spectrum_2d=fn)
+
+    pext = specviz2d_helper.app.get_tray_item_from_name('spectral-extraction')
+    assert pext.bg_type_selected == 'OneSided'
+    assert pext.bg_separation < 0
```

Comment:
What is `< 0` testing? You sure you don't want to be more explicit with `numpy.testing.assert_allclose()`?

Reply:
It's really just testing that the logic detected that the extraction window was so close to the top of the image, that the background region needed to be below the extraction region.  I was intentionally non-specific in the test, because we'll very likely improve the separation/width default logic in the future which would change the exact value, but as long as the window-detection logic doesn't change significantly to avoid the edge entirely, that should still always return a negative separation.